Firestore/core/test/firebase/firestore/local/mutation_queue_test.cc

```diff
@@ -0,0 +1,432 @@
+      ASSERT_EQ(found->batch_id(), next.batch_id());
+    }
+
+    const MutationBatch& first = batches[0];
+    absl::optional<MutationBatch> found =
+        mutation_queue_->NextMutationBatchAfterBatchId(first.batch_id() - 42);
+    ASSERT_EQ(found->batch_id(), first.batch_id());
+
+    const MutationBatch& last = batches[batches.size() - 1];
+    absl::optional<MutationBatch> not_found =
+        mutation_queue_->NextMutationBatchAfterBatchId(last.batch_id());
+    ASSERT_EQ(not_found, absl::nullopt);
+  });
+}
+
+TEST_P(MutationQueueTest, AllMutationBatchesAffectingDocumentKey) {
+  persistence_->Run("test_all_mutation_batches_affecting_document_key", [&] {
+    std::vector<Mutation> mutations = {
+        testutil::SetMutation("foi/bar", Map("a", 1)),
+        testutil::SetMutation("foo/bar", Map("a", 1)),
+        testutil::PatchMutation("foo/bar", Map("b", 1), {}),
+        testutil::SetMutation("foo/bar/suffix/key", Map("a", 1)),
+        testutil::SetMutation("foo/baz", Map("a", 1)),
+        testutil::SetMutation("food/bar", Map("a", 1)),
+    };
+
+    // Store all the mutations.
+    std::vector<MutationBatch> batches;
+    for (const Mutation& mutation : mutations) {
+      MutationBatch batch =
+          mutation_queue_->AddMutationBatch(Timestamp::Now(), {}, {mutation});
+      batches.push_back(batch);
+    }
+
+    std::vector<MutationBatch> expected{batches[1], batches[2]};
+    std::vector<MutationBatch> matches =
+        mutation_queue_->AllMutationBatchesAffectingDocumentKey(
+            testutil::Key("foo/bar"));
+
+    ASSERT_EQ(matches, expected);
+  });
+}
+
+TEST_P(MutationQueueTest, AllMutationBatchesAffectingDocumentKeys) {
+  persistence_->Run("test_all_mutation_batches_affecting_document_key", [&] {
+    std::vector<Mutation> mutations = {
+        testutil::SetMutation("fob/bar", Map("a", 1)),
+        testutil::SetMutation("foo/bar", Map("a", 1)),
+        testutil::PatchMutation("foo/bar", Map("b", 1), {}),
+        testutil::SetMutation("foo/bar/suffix/key", Map("a", 1)),
+        testutil::SetMutation("foo/baz", Map("a", 1)),
+        testutil::SetMutation("food/bar", Map("a", 1)),
+    };
+
+    // Store all the mutations.
+    std::vector<MutationBatch> batches;
+    for (const Mutation& mutation : mutations) {
+      MutationBatch batch =
+          mutation_queue_->AddMutationBatch(Timestamp::Now(), {}, {mutation});
+      batches.push_back(batch);
+    }
+
+    DocumentKeySet keys{
+        Key("foo/bar"),
+        Key("foo/baz"),
+    };
+
+    std::vector<MutationBatch> expected{batches[1], batches[2], batches[4]};
+    std::vector<MutationBatch> matches =
+        mutation_queue_->AllMutationBatchesAffectingDocumentKeys(keys);
+
+    ASSERT_EQ(matches, expected);
+  });
+}
+
+TEST_P(MutationQueueTest,
+       AllMutationBatchesAffectingDocumentKeys_handlesOverlap) {
+  persistence_->Run(
+      "test_all_mutation_batches_affecting_document_keys_handlesOverlap", [&] {
```

Comment:
Ultranit: s/handlesOverlap/handles_overlap/?

Reply:
Fixed all these strings. My conversion script botched these. They really should match the name of the test case.